### DISCUSSION TITLE
Which Version for Java8

### DISCUSSION THREAD

**Discussion by punitsa** [oracle/opengrok#4190] [Q&A] [answered]
Hi.

which is the last version that supports Java 8 / tomcat 9 ?

Thanks

**Comment by vladak** [comment, marked as answer]
The switch to Java 11 was done in 1.5.0, so 1.4.9 was the last one in this regard. You'd be far better of with running up-to-date version, say in a Docker container or use some other virtualized environment.

**Reply by punitsa** [reply]
I agree and understand. but I have rhel6 box and it is hard to find rpms for it. but let me explore docker option.

**Reply by vladak** [reply]
That is sort of similar to my situation. I maintain a production instance running Solaris 11 and for that I had to actually build Java 11 (incrementally from Java 8 source, version by version) and bring Tomcat 10. It works fine.

**Reply by vladak** [reply]
Also, let's face it: with such old version you will run into a bug fixed in newer versions rather sooner than later. And given that only selected few issues are being fixed for the current release, you will be completely on your own with older releases.

**Reply by punitsa** [reply]
Thanks. I was able to get JDK 11 installed and it went fine. One comment, I was not able to make (very old version of autoconf) universal ctags on my machine. but opengrok did work correctly with exerbutant ctags. I thought universal ctags was mandatory.

**Reply by vladak** [reply]
It does require Universal ctags. Moreover, in the recent versions it also enforces working Universal ctags by running it against a known file created under source root, so I guess it depends on your interpretation/observation of what "working" means.
